packages/mail/src/classes/mail-service.js

```diff
@@ -184,6 +184,15 @@ class MailService {
         data.substitutionWrappers = this.substitutionWrappers;
       }
 
+      //Tests to make sure the email will not crash when sent
+      if ('text' in data && 'dynamicTemplateData' in data) {
+        throw new Error ('text and dynamicTemplateData fields cannot both have values');
+      }
+
+      if ('html' in data && 'dynamicTemplateData' in data) {
+        throw new Error ('html and dynamicTemplateData fields cannot both have values');
+      }
+
```

Comment:
This check would be better located in the implementation for [`addTextContent`](https://github.com/sendgrid/sendgrid-nodejs/blob/main/packages/helpers/classes/mail.js#L399) and [`addHtmlContent`](https://github.com/sendgrid/sendgrid-nodejs/blob/main/packages/helpers/classes/mail.js#L415):
```
if (typeof this.dynamicTemplateData !== 'undefined') {
  throw new Error('text and dynamicTemplateData fields cannot both have values');
}
```
Those functions get called downstream from [`Mail.create(data)`](https://github.com/sendgrid/sendgrid-nodejs/pull/1197/files#diff-f146eb96649a37994e2da1d195d60789R197).